From ea64cad247465aec7cb773506a5a303d03c9cba6 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 14 Mar 2018 01:05:37 +0530
Subject: [PATCH 0340/1852] v4l: xilinx: sdirxss: Correct the v4l2_field to
 V4L2_FIELD_ALTERNATE

commit a3be715bf8be9cc48632424f41ca60a63389f0d2 from
https://github.com/Xilinx/linux-xlnx.git

Correct the field type to V4L2_FIELD_ALTERNATE instead of
V4L2_FIELD_INTERLACED and halve the height in such cases.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index 2d66c1e05345..1d360d1592b8 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -669,7 +669,7 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 						format->field = V4L2_FIELD_NONE;
 					else
 						format->field =
-							V4L2_FIELD_INTERLACED;
+							V4L2_FIELD_ALTERNATE;
 				} else {
 					format->width = 1920;
 					format->height = 1080;
@@ -677,7 +677,7 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 						format->field = V4L2_FIELD_NONE;
 					else
 						format->field =
-							V4L2_FIELD_INTERLACED;
+							V4L2_FIELD_ALTERNATE;
 				}
 				break;
 			case XSDIRX_TS_DET_STAT_RATE_50HZ:
@@ -720,7 +720,7 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 		}
 		break;
 	case XSDIRX_MODE_SD_MASK:
-		format->field = V4L2_FIELD_INTERLACED;
+		format->field = V4L2_FIELD_ALTERNATE;
 
 		switch (family) {
 		case XSDIRX_NTSC:
@@ -804,9 +804,12 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 		if (pic_type)
 			format->field = V4L2_FIELD_NONE;
 		else
-			format->field = V4L2_FIELD_INTERLACED;
+			format->field = V4L2_FIELD_ALTERNATE;
 	}
 
+	if (format->field == V4L2_FIELD_ALTERNATE)
+		format->height = format->height / 2;
+
 	switch (sampling) {
 	case XST352_BYTE3_COLOR_FORMAT_420:
 		format->code = MEDIA_BUS_FMT_VYYUYY8_1X24;
-- 
2.31.1

